doc/GLM-introduction.rst

```diff
@@ -0,0 +1,435 @@
+===================
+Introduction to GLM
+===================
+
+In Generalized Linear Model (GLM), we mainly want to solve the following problem.
+
+.. math::
+
+    \min_{\beta_0, \beta} \frac{1}{N} \sum_{i = 1}^N w_i \ell (y_i, \beta_0 + \beta^T x_i)
+    + \lambda [0.5(1 - \alpha)\| \beta \|_2^2 + \alpha \| \beta \|_1]
+
+where :math:`\ell (y_i, \beta_0 + \beta^T x_i)` is negative log-likelihood of
+an observation :math:`i`. We explain details of GLM and how we optimize
+the cost function more in introduction to GLM.
+
+
+GLM with elastic net penalty
+----------------------------
+
+**Poisson-like GLM**
+
+The ``pyglmnet`` implementation comes with `poisson`, `poissonexp`, `binomial`,
+`normal` and `multinomial` distributions, but for illustration, I will walk you
```

Comment:
should we change the `I` to `we` in the documentation? @pavanramkumar okay with it?